docs/data/data-grid/editing-legacy/editing.md

```diff
@@ -0,0 +1,272 @@
+When the callback is called, use the API to check the value of the other field and decide if the current value is valid or not.
+Return a new object contaning `params.props` and the `error` attribute with the validation status.
+Once at the least one field has the `error` attribute equals to true no new value will be committed.
+
+**Note:** For server-side validation, the same [approach](#server-side-validation) from the cell editing can be used. The data grid will wait for all promises to resolve before commiting.
+
+The following demo requires a value for the "Payment method" column if the "Is paid?" column was checked.
+
+{{"demo": "ConditionalValidationGrid.js", "disableAd": true, "bg": "inline", "defaultCodeOpen": false}}
+
+> The conditional validation can also be implemented with the [controlled editing](#controlled-editing-2).
+> This approach can be used in the free version of the DataGrid.
+> The only limitation is that it does not work with the `singleSelect` column type.
+
+### Control with external buttons [<span class="plan-pro"></span>](https://mui.com/store/items/material-ui-pro/)
+
+You can [disable the default behavior](/components/data-grid/events/#disabling-the-default-behavior) of the grid and control the row edit using external buttons.
+
+Here is shown how a full-featured CRUD can be created.
+
+{{"demo": "FullFeaturedCrudGrid.js", "bg": "inline", "disableAd": true}}
+
+### Saving rows with nested structures
+
+Saving columns that make use of `valueGetter` can be done adding a `valueSetter`.
+The same [approach](/components/data-grid/editing/#saving-nested-structures) from the cell editing mode can be used here.
+Note that the `valueSetter` will be called for each field.
+
+### Server-side persistence [<span class="plan-pro"></span>](https://mui.com/store/items/material-ui-pro/)
+
+If you want to send the updated data to your server, you can use the `onRowEditCommit` which is fired just before committing the new cell value to the grid.
+
+To access the new values for the row, use `apiRef.current.getEditRowsModel` to get all rows in edit mode, then use the id provided to get only the values for the row that was committed.
+
+You can then decide if you want to send the whole row or only the modified fields, by checking them against the previous row values.
+
+{{"demo": "RowEditServerSidePersistence.js", "disableAd": true, "bg": "inline", "defaultCodeOpen": false}}
+
+### Events [<span class="plan-pro"></span>](https://mui.com/store/items/material-ui-pro/)
+
+The following events can be imported and used to customize the row edition:
+
+- `rowEditStart`: emitted when the row turns to edit mode.
+- `rowEditStop`: emitted when the row turns back to view mode.
+- `rowEditCommit`: emitted when the new row values are committed.
+- `editCellPropsChange`: emitted when the props passed to an edit cell component are changed.
```

Comment:
- `rowEditStart`: triggered when a row enters edit mode.
- `rowEditStop`: triggered when a row returns to view mode.
- `rowEditCommit`: triggered when new row values are committed.
- `editCellPropsChange`: triggered when the props passed to the edit cell component are changed.